sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureApplicationCredentialBuilder.java

```diff
@@ -67,6 +69,22 @@ public AzureApplicationCredentialBuilder managedIdentityResourceId(String resour
         return this;
     }
 
+    /**
+     * Specifies the object ID of user assigned or system assigned identity, when this credential is running
+     * in an environment with managed identities. If unset, the value in the AZURE_CLIENT_ID environment variable
+     * will be used. If neither is set, the default value is null and will only work with system assigned
+     * managed identities and not user assigned managed identities.
+     *
+     * Only one of managedIdentityResourceId and managedIdentityClientId can be specified.
+     *
+     * @param objectId the object ID
+     * @return the DefaultAzureCredentialBuilder itself
+     */
+    public AzureApplicationCredentialBuilder managedIdentityObjectId(String objectId) {
+        this.managedIdentityObjectId = objectId;
+        return this;
+    }
```

Comment:
I kind of suspect this is the same bucket as the `DefaultAzureCredential` - I think our idea was that if you think you need `objectId` support you are in a very narrow case and you just make a `ManagedIdentityCredential` directly.

Reply:
scoped to MI cred only.